modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -128,5 +132,51 @@ object ConstructionService {
           }
         }
 
+    override def getPayloads(
+      operations: NonEmptyList[Operation],
+      maybeMetadata: Option[TransactionMetadata]
+    ): EitherT[F, ConstructionError, PayloadsResult] =
+      for {
+        _ <- EitherT.cond(operations.size == 2, (), IncorrectNumberOfOperations)
+
+        (positiveOperation, transactionAmount) <- EitherT.fromOption(
+          operations.collectFirst {
+            case op if op.amount.exists(!_.value.isNegative) =>
+              op.amount
+                .flatMap(_.value.toTransactionAmount.toOption)
+                .map((op, _))
+          }.flatten,
+          MissingPositiveAmountOperation
```

Comment:
imo you should verify that positive operation has the same balance (positive balance) as the negative operation (negative balance)